docs/proposals/11-scheduling.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Reservation-Based Scheduling
+
+iep-number: 11
+
+creation-date: 2024-09-30
+
+status: draft
+
+authors:
+
+- "@lukasfrank"
+
+
+reviewers:
+
+- "@afritzler"
+- "@balpert89"
+
+---
+
+# IEP-11: Reservation-Based Scheduling
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+Scheduling resources is a central process to increase utilization of cloud infrastructure. It not only involves the 
+process to find any feasible `MachinePool` but rather, if possible, to find a `MachinePool` where the `Machine` can 
+be materialized.
+
+
+## Motivation
+
+With `ironcore` a multi-level hierarchy can be built to design availability/failure zones. `Machine`s can be created 
+on every layer, which are subsequently scheduled on `MachinePool`s  in the hierarchy below.  A scheduler on every 
+layer should assign a `Machine` on a `MachinePool` which can fulfill it.  This proposal addresses the known 
+limitations of the current implementation:
```

Comment:
No. 1 is a limitation No. 2 is an implementation detail which we have. The biggest limitation however is that we can not guarantee at the moment that we find enough resources on a compute pool and blindly schedule `Machine` objects onto pools even though they might not be fullfilled.

Reply:
> No. 2 is an implementation detail which we have 
which is a limitation since we at the moment only support CPU and memory